cmd/mmlogicapi/apisrv/apisrv.go

```diff
@@ -0,0 +1,659 @@
+				Roster:  &partialRoster,
+			}
+			if err = stream.Send(poolChunk); err != nil {
+				stats.Record(fnCtx, MlGrpcErrors.M(1))
+				return err
+			}
+			partialRoster.Players = []*mmlogic.Player{}
+		}
+
+		// Add one additional player result to the partial pool.
+		player := &mmlogic.Player{Id: playerList[i], Attributes: []*mmlogic.Player_Attribute{}}
+		// Collect all the filtered attributes into the player protobuf.
+		for attribute, fr := range filteredResults {
+			if value, ok := fr[playerList[i]]; ok {
+				player.Attributes = append(player.Attributes, &mmlogic.Player_Attribute{Name: attribute, Value: value})
+			}
+		}
+		partialRoster.Players = append(partialRoster.Players, player)
+
+	}
+
+	mlLog.WithFields(log.Fields{"count": len(playerList), "pool": pool.Name}).Debug("player pool streaming complete")
+
+	stats.Record(fnCtx, MlGrpcRequests.M(1))
+	return nil
+}
+
+// applyFilter is a sequential query of every entry in the Redis sorted set
+// that fall beween the minimum and maximum values passed in through the filter
+// argument.  This can be likely sped up later using concurrent access, but
+// with small enough player pools (less than the 'redis.queryArgs.count' config
+// parameter) the amount of work is identical, so this is fine as a starting point.
+// If the provided field is not indexed or the provided range is too large, a nil result
+// is returned and this filter should be disregarded when applying filter overlaps.
+func (s *mmlogicAPI) applyFilter(c context.Context, filter *mmlogic.Filter) (map[string]int64, error) {
+	type pName string
+
+	// Default maximum value is positive infinity (i.e. highest possible number in redis)
+	// https://redis.io/commands/zrangebyscore
+	maxv := strconv.FormatInt(filter.Maxv, 10) // Convert int64 to a string
+	if filter.Maxv == 0 {                      // No max specified, set to +inf
+		maxv = "+inf"
+	}
+
+	mlLog.WithFields(log.Fields{"filterField": filter.Attribute}).Debug("In applyFilter")
+
+	// Get redis connection from pool
+	redisConn := s.pool.Get()
+	defer redisConn.Close()
+
+	// Check how many expected matches for this filter before we start retrieving.
+	cmd := "ZCOUNT"
+	count, err := redis.Int64(redisConn.Do(cmd, filter.Attribute, filter.Minv, maxv))
+	//DEBUG: count, err := redis.Int64(redisConn.Do(cmd, "BLARG", filter.Minv, maxv))
+	mlLog := mlLog.WithFields(log.Fields{
+		"query": cmd,
+		"field": filter.Attribute,
+		"minv":  filter.Minv,
+		"maxv":  maxv,
+		"count": count,
+	})
+	if err != nil {
+		mlLog.WithFields(log.Fields{"error": err.Error()}).Error("state storage error")
+		return nil, err
+	}
+
+	if count == 0 {
+		err = errors.New("filter applies to no players")
+		mlLog.Error(err.Error())
+		return nil, err
+	} else if count > 500000 {
+		// 500,000 results is an arbitrary number; OM doesn't encourage
+		// patterns where MMFs look at this large of a pool.
+		mlLog.Warn("filter applies to too many players, ignoring")
+		return nil, nil
```

Comment:
I feel like Warn should be Error if we don't intend on doing the thing that was asked for

Reply:
Yep.